### PR TITLE
global: invenio helm charts for openshift

### PR DESCRIPTION
**Pull request by ppanero**
Helm Charts for OpenShift deployment.

It has been tested with an instance from Invenio and another of InvenioRDM. Note that the configured docker image does not exist.

Table of contents:

- `benchmark`: This folder contains the files in order to perform a load/stress test. A `cli.py` file for batch record creation and a `locust.py` file with the tests to perform. Instructions are available in this folder `README.md` file.
- `invenio`: The folder with the actual Helm charts.
- `jobs`: Contains the helm charts to create ad-hoc jobs and cronjobs. They are not part of the instance charts (`invenio` folder) since otherwise a job would be created when installing and this are only needed in certain cases, and they are all different (different command).
- `README.md`: Contains instructions on how to use/install an Invenio instance using this charts
- `PUBLISHING.md`: Contains instructions on how to publish new helm charts.

Note: The repository is not per se available in the GH pages of this repo, it has been tested in an external one. Once this is merge it will be deployed here too (How to do so are the instructions in the `PUBLISHING.md`).

### CODE REVIEW

File: PUBLISHING.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+``` console
+$ cr upload -o inveniosoftware -r helm-invenio -p .deploy -t  <TOKEN>
+```
+
+:warning: the `.deploy` folder should only contain new releases, otherwise chart-releaser will try to push them all and will fail due to "already existing" versions.
+
+And finally we should update the chart `index.yaml` file, that will be served to `helm`:
+
+``` console
+$ cr index -i ./index.yaml -p .deploy/ -o inveniosoftware -c https://github.com/inveniosoftware/helm-invenio -r helm-invenio --token <TOKEN>
```
Comment (ntarocco) [inline PUBLISHING.md]: what is `cr`? `chart releaser`?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+$ oc create secret generic \
+  --from-literal="ELASTICSEARCH_PASSWORD=$ELASTICSEARCH_PASSWORD" \
+  --from-literal="ELASTICSEARCH_USER=$ELASTICSEARCH_USER" \
+  elasticsearch-secrets
+```
+
+:warning: For now the previous Elasticsearch environment variables are not ported to `invenio-search`, therefore the way to create the secret is:
+
+``` console
+$ export INVENIO_SEARCH_ELASTIC_HOSTS="[{'host': 'es-cdsbooks.cern.ch', 'url_prefix': '/es', 'timeout': 30, 'port': 443, 'use_ssl': True, 'verify_certs': False, 'http_auth':('USERNAME_CHANGEME', 'PASSWORD_CHANGEME')}]"
```
Comment (ntarocco) [inline README.md]: `es-cdsbooks.cern.ch` -> `myes.domain.org` ?

File: PUBLISHING.md (hunk 1 above)
Comment (ppanero) [reply PUBLISHING.md]: yes, I'll put a note :)

File: README.md (hunk 2 above)
Comment (ppanero) [reply README.md]: yeah, sorry about that. Thought I took it out :)

**Comment by ppanero** [comment]
The master branch is reflecting this commits, I am not sure why it did not got automatically closed.

**Pull request closed without merging**